tests/test_gds_metrics.py

```diff
@@ -75,6 +86,135 @@ def test_update(self):
+        preds = np.array([1, 0, 1])
+        truth = np.array([1, 1, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0.5)
+
+class TestGDSMSE(unittest.TestCase):
+    def test_init(self):
+        measure = MSE()
+        self.assertIsNone(measure.value)
+
+    def test_update(self):
+        measure = MSE()
+        preds = np.array([1, 0.5])
+        truth = np.array([1, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0.125)
+        preds = np.array([1, 0, 0.5])
+        truth = np.array([1, 0.5, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0.15)
+
+class TestGDSRMSE(unittest.TestCase):
+    def test_init(self):
+        measure = RMSE()
+        self.assertIsNone(measure.value)
+
+    def test_update(self):
+        measure = RMSE()
+        preds = np.array([1, 0.5])
+        truth = np.array([1, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0.125**.5)
+        preds = np.array([1, 0, 0.5])
+        truth = np.array([1, 0.5, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0.15**.5)
+
+
+class TestGDSMAE(unittest.TestCase):
+    def test_init(self):
+        measure = MAE()
+        self.assertIsNone(measure.value)
+
+    def test_update(self):
+        measure = MAE()
+        preds = np.array([1, 0.5])
+        truth = np.array([1, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0.25)
+        preds = np.array([1, 0, 0.5])
+        truth = np.array([1, 0.5, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0.3)
+
+class TestGDSHitsAtK(unittest.TestCase):
+    def test_init(self):
+        measure = HitsAtK(k=1)
+        self.assertIsNone(measure.value)
+
+    def test_update(self):
+        measure = HitsAtK(k=1)
+        preds = np.array([0.2, 0.5])
+        truth = np.array([1, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0)
+        preds = np.array([0.7, 0.1, 0.5])
+        truth = np.array([1, 0.5, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0.5)
+
+class TestGDSRecallAtK(unittest.TestCase):
+    def test_init(self):
+        measure = RecallAtK(k=1)
+        self.assertIsNone(measure.value)
+
+    def test_update(self):
+        measure = RecallAtK(k=1)
+        preds = np.array([0.2, 0.5])
+        truth = np.array([1, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 0)
+        preds = np.array([0.7, 0.1, 0.5])
+        truth = np.array([1, 1, 0])
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 1/3)
+
+class TestGDSConfusionMatrix(unittest.TestCase):
+    def test_init(self):
+        measure = ConfusionMatrix(num_classes=2)
+        self.assertIsNone(measure.value)
+
+    def test_update(self):
+        measure = ConfusionMatrix(num_classes=2)
```

Comment:
Do we have test for num_classes>2 for any of the the ConfusionMatrix related metrics?